### ISSUE TITLE
Microsoft Dynamics 365 CRM field mapping and RequiredLevel

### ISSUE THREAD

**Issue by jamesmacwhite** [verbb/formie#1249]
### Describe the bug

I'm looking at the entity mapping between our forms and Microsoft Dynamics 365 CRM system. I've noticed that Formie seems to be marking fields that are set to the RequiredLevel of Recommended as mandatory in the model.

For example, we have a field on the contact entity called `ccl1000_studentnumber` (Student Number). The full definition is below for reference.

```json
{
	"@odata.type": "#Microsoft.Dynamics.CRM.StringAttributeMetadata",
	"AttributeType": "String",
	"IsCustomAttribute": true,
	"IsValidForCreate": true,
	"IsValidForUpdate": true,
	"CanBeSecuredForCreate": true,
	"CanBeSecuredForUpdate": true,
	"LogicalName": "ccl1000_studentnumber",
	"SchemaName": "ccl1000_StudentNumber",
	"MetadataId": "21902777-f3fd-4598-ba71-afd8c6fd6d0c",
	"DisplayName": {
		"LocalizedLabels": [
			{
				"Label": "Student Number",
				"LanguageCode": 1033,
				"IsManaged": true,
				"MetadataId": "8cd69e92-6d2e-47e0-ab92-c2175152f2e6",
				"HasChanged": null
			}
		],
		"UserLocalizedLabel": {
			"Label": "Student Number",
			"LanguageCode": 1033,
			"IsManaged": true,
			"MetadataId": "8cd69e92-6d2e-47e0-ab92-c2175152f2e6",
			"HasChanged": null
		}
	},
	"RequiredLevel": {
		"Value": "Recommended",
		"CanBeChanged": true,
		"ManagedPropertyLogicalName": "canmodifyrequirementlevelsettings"
	}
}
```

It's required level is set to Recommended. Based on the documentation for the various level values: https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/reference/attributerequiredlevel?view=dataverse-latest, I would suggest that SystemRequired and ApplicationRequired should potentially be enforced in the model, but not fields set to "Recommended" as by the documentation, it wouldn't suggest this type should be mandatory.

Currently, it looks like recommended fields are being forced as mandatory, when they aren't in fact required by the CRM system.

![image](https://user-images.githubusercontent.com/8067792/213860446-443d6eb3-8b77-4298-b49d-17073fd37668.png)

Would it be possible to remove the required status in the model on "Recommended" or perhaps update the indicator to a blue icon (this is what CRM as an application would do).

### Steps to reproduce

1. Refresh Microsoft Dynamics 365 integration for field mapping
2. Note this field is being marked mandatory for a recommended status

### Form settings

N/A

### Craft CMS version

3.7.63.1

### Plugin version

1.6.21

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.6.21",
  "...": "..."
}
```

Then run `composer update`.

**Comment by jamesmacwhite** [comment]
I would suggest potentially adjusting this line:

https://github.com/verbb/formie/blob/craft-3/src/integrations/crm/MicrosoftDynamics365.php#L443

```php
'required' => ($requiredLevel === 'None' || $requiredLevel === 'Recommended') ? false : true,
```

**Comment by jamesmacwhite** [comment]
Ha, beat me to it!

Thank you, this will help. We have found our CRM has some interesting fields that are being forced in the model, which don't necessarily need to be.

**Comment by engram-design** [comment]
Fixed in [1.6.22](https://github.com/verbb/formie/releases/tag/1.6.22)
